data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/MutateMapper.java

```diff
@@ -0,0 +1,104 @@
+package org.opensearch.dataprepper.logstash.mapping;
+
+import com.amazon.dataprepper.model.configuration.PluginModel;
+import org.opensearch.dataprepper.logstash.model.LogstashAttribute;
+import org.opensearch.dataprepper.logstash.model.LogstashPlugin;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class MutateMapper {
+    public static class AddEntryConfig {
+        public final String key;
+        public final Object value;
+
+        public AddEntryConfig(String key, Object value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
+
+    public static class RenameCopyConfig {
+        public final String from_key;
+        public final String to_key;
+
+        public RenameCopyConfig(String fromKey, String toKey) {
+            this.from_key = fromKey;
+            this.to_key = toKey;
+        }
+    }
+
+    public List<PluginModel> getModels(LogstashPlugin logstashPlugin) {
+        List<PluginModel> models = new LinkedList<>();
+        List<AddEntryConfig> adds = new LinkedList<>();
+        List<RenameCopyConfig> renames = new LinkedList<>();
+        List<ArrayList<String>> deletes = new LinkedList<>();
+        List<RenameCopyConfig> copies = new LinkedList<>();
+
+        for(LogstashAttribute attr : logstashPlugin.getAttributes()) {
+            final String name = attr.getAttributeName();
+            if(Objects.equals(name, "add_field")) {
+                ((Map<String, Object>)attr.getAttributeValue().getValue()).entrySet().forEach(entry -> {
+                    adds.add(new AddEntryConfig(entry.getKey(), entry.getValue()));
+                });
+            } else if(Objects.equals(name, "rename")) {
+                ((Map<String, String>)attr.getAttributeValue().getValue()).entrySet().forEach(entry -> {
+                    renames.add(new RenameCopyConfig(entry.getKey(), entry.getValue()));
+                });
+            } else if(Objects.equals(name, "remove_field")) {
+                deletes.add((ArrayList<String>)attr.getAttributeValue().getValue());
+            } else if(Objects.equals(name, "copy")) {
+                ((Map<String, String>)attr.getAttributeValue().getValue()).entrySet().forEach(entry -> {
+                    copies.add(new RenameCopyConfig(entry.getKey(), entry.getValue()));
+                });
+            }
+        }
+
+        if(adds.size() > 0) {
```

Comment:
Are these being placed in the right order?

I see from the Logstash documentation that `rename` comes before `copy`. But, where do `add_field` and `remove_field` come in?

Reply:
I'll reorder it to the way Logstash has it set up. `add_field` and `remove_field` are at the bottom of the documentation but I believe it can go wherever